esphome/components/honeywellabp2_i2c/honeywellabp2.cpp

```diff
@@ -0,0 +1,109 @@
+#include "honeywellabp2.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace honeywellabp2_i2c {
+
+static const uint8_t STATUS_BIT_POWER = 6;
+static const uint8_t STATUS_BIT_BUSY = 5;
+static const uint8_t STATUS_BIT_ERROR = 2;
+static const uint8_t STATUS_MATH_SAT = 0;
+
+static const char *const TAG = "honeywellabp2";
+
+void HONEYWELLABP2Sensor::setup() { ESP_LOGD(TAG, "Setting up Honeywell ABP2 Sensor "); }
+
+void HONEYWELLABP2Sensor::read_sensor_data() {
+  if (this->read(raw_data_, 7) != i2c::ERROR_OK) {
+    ESP_LOGE(TAG, "Communication with ABP2 failed!");
+    this->mark_failed();
+    return;
+  }
+  float press_counts = raw_data_[3] + raw_data_[2] * 256 + raw_data_[1] * 65536;  // calculate digital pressure counts
+  float temp_counts = raw_data_[6] + raw_data_[5] * 256 + raw_data_[4] * 65536;  // calculate digital temperature counts
+
+  this->last_pressure_ = ((press_counts - this->min_count_) * (double) (this->max_pressure_ - this->min_pressure_)) /
```

Comment:
I could be wrong, but this cast should not be needed and we try to avoid `double` in ESPHome.
```suggestion
  this->last_pressure_ = ((press_counts - this->min_count_) * (this->max_pressure_ - this->min_pressure_)) /
```